js/src/Ice/ConnectionI.js

```diff
@@ -1166,63 +1154,46 @@ export class ConnectionI {
     }
 
     validate() {
-        if (this._adapter !== null) {
```

Comment:
This was checking if it was an incoming or outgoing connection. For JavaScript, we know it is always an outgoing connection.

Reply:
We did not check `this._incoming` but `this._adapter`. An outgoing connection can have an associated object adapter.

I assume the application can't get hold of the connection and call setAdapter before the call to `validate()`?